server/client/ruby/candlepin_api.rb

```diff
@@ -809,7 +809,7 @@ def create_product(owner_key, id, name, params={})
       'dependentProductIds' => dependentProductIds,
       'branding' => branding,
       'reliesOn' => relies_on,
-      'providedProducts' => providedProducts.collect { |pid| {'id' => pid} },
+      'providedProducts' => providedProducts.collect { |pid| {'id' => pid} || pid}
```

Comment:
This is added because some places provided products are used as object or string(id)

Reply:
The ruby stuff is on life-support at this point, but we should try to avoid conflating them. If we're not consistent in how we pass provided products, we should spend a bit investigating how much would it would be to fix that inconsistency, rather than codifying it. From there we can decide if it's low enough effort to fix properly, or leaving something like this (along with a comment explaining it).

Reply:
I recollect this was added because I was passing the whole provided-product object. After discussion with @abhiskum, I understood this will work even if we just pass a string of Ids. 
Hence now removed passing whole object to `create_product`.